addons/payment_custom/models/payment_provider.py

```diff
@@ -51,7 +51,7 @@ def action_recompute_pending_msg(self):
                     f'</div>'
 
     @api.model
-    def _get_removal_domain(self, provider_code, custom_mode='', **kwargs):
+    def _get_removal_domain(self, provider_code, *, custom_mode='', **kwargs):
```

Comment:
I don't see what you intend to fix here?

Reply:
A limitation of the linter, it sees this as a possibility of an error when calling `_get_removal_domain(provider_code, 'xxx')` which is not supported in the parent function.

Reply:
I get your point but I would rather not change the method signature to tolerate what I see as a basic programming mistake that can be fixed instantly.

Reply:
Why fix when you can prevent?

Reply:
The way I see it, we're adding more code to allow bad code.

I'd rather keep things as they are now since I can't recall having to merge a single bugfix because of that. Instead, we could enforce that keyword arguments are passed with their key name.

Reply:
By the way, this is a feature that can already be used in IDEs
![image](https://github.com/user-attachments/assets/cf2d0e86-8697-4155-b53a-c1517ff80a85)

My IDE is a bit more flexible than this test though, as you can see I can add new arguments without complaints, even without making them kw-only
![image](https://github.com/user-attachments/assets/d8749b25-c062-4b53-8494-d88cfcc54fe5)

Reply:
TBH, this is not a hill I'd die on, but I'm not comfortable with a change that I see as random. It's making some methods more strict and some others less strict, while most methods that should also be impacted are left alone. I fear that the place to look for reasons why an override was implemented one way or another will be this PR's discussion history.